client/network/sync/src/lib.rs

```diff
@@ -576,14 +576,32 @@ where
 			.map(|p| PeerInfo { best_hash: p.best_hash, best_number: p.best_number })
 	}
 
+	/// Returns the best seen block.
+	fn best_seen(&self) -> Option<NumberFor<B>> {
+		let mut best_seens = self.peers.values().map(|p| p.best_number).collect::<Vec<_>>();
+		best_seens.sort_unstable();
```

Comment:
No need to sort the whole thing.  [select_nth_unstable](https://doc.rust-lang.org/std/primitive.slice.html#method.select_nth_unstable)

Reply:
Done and learned something new :)